bindings/azure/openai/openai.go

```diff
@@ -181,13 +197,27 @@ func (p *AzOpenAI) Invoke(ctx context.Context, req *bindings.InvokeRequest) (res
 		resp.Data = responseAsBytes
 
 	case ChatCompletionOperation:
+		if p.deploymentID == "" {
+			return nil, fmt.Errorf("required metadata not set: %s", DeploymentID)
+		}
 		response, err := p.chatCompletion(ctx, req.Data, req.Metadata)
 		if err != nil {
 			return nil, fmt.Errorf("error performing chat completion: %w", err)
 		}
 		responseAsBytes, _ := json.Marshal(response)
 		resp.Data = responseAsBytes
 
+	case GetEmbeddingOperation:
+		if p.embeddingDeploymentID == "" {
```

Comment:
Should we make these properties something that's passed in each request then maybe? Rather than being hardcoded on the component

Reply:
That's a good idea. Considering we can very well have multiple models for multiple use cases which converts to different calls to the binding

Reply:
Updated.